### PR TITLE
Remove rtm.start and start_method configuration

### PR DESCRIPTION
**Pull request by kstole**
Now that the `rtm.start` method has been deprecated (and has a response matching `rtm.connect`), there is only one method for connecting to the Real Time Messaging API.

### CODE REVIEW

**Comment by kstole** [comment]
We get a lot of changes by running the API update tasks, so I left those changes out of this and just removed the `#rtm_start` method manually

File: .rubocop.yml [hunk 1]
```diff
@@ -35,6 +35,10 @@ Style/Documentation:
 Style/ModuleFunction:
   Enabled: false
 
+Style/NumericLiterals:
+  Exclude:
+    - 'spec/**/*'
```
Comment (kstole) [inline .rubocop.yml:40]: I'm not sure the best way to disable this just for Unix timestamps so I just disabled it entirely for specs

**Comment by dblock** [comment]
Good stuff, thanks!

**Comment by dblock** [comment]
Added you here too to maintainers (https://github.com/slack-ruby/slack-ruby-client/invitations), thanks for picking up the overdue update tasks @kstole!

**Pull request merged**